src/controllers/EmojiController.php

```diff
@@ -0,0 +1,255 @@
+     * @param ResponseInterface      ResponseInterface      $response PSR-& standard for sending server response
+     *
+     * @return ResponseInterface HTTP response of client request
+     */
+    public function index(ServerRequestInterface $request, ResponseInterface $response)
+    {
+        //Get all Emojis and their associated keywords
+        $emojis = Emoji::with('keywords')->get();
+
+        //Return Data in array
+        $result = [];
+        foreach ($emojis as $emoji) {
+            $result[] = $this->buildEmojiData($emoji);
+        }
+
+        //Write to response output
+        $response->getBody()->write(json_encode($result));
+
+        //Return response
+        return $response;
+    }
+
+    /**
+     * GETs a single Emoji.
+     *
+     * @param ServerRequestInterface ServerRequestInterface $request  PSR-7 standard for receiving client request
+     * @param ResponseInterface      ResponseInterface      $response PSR-& standard for sending server response
+     *
+     * @return ResponseInterface HTTP response of client request
+     */
+    public function show(ServerRequestInterface $request, ResponseInterface $response, $argc)
+    {
+        //Get the single Emoji with the id from route
+        $emoji = Emoji::find($argc['id']);
+
+        //Check if the Emoji with the id exist
+        if ($emoji != null) {
+            //Build Emoji data to return
+            $result = $this->buildEmojiData($emoji);
+        } else {
+            //Repond with 404 if Emoji not found
+            $response = $response->withStatus(404);
+            $result = ['message' => 'Emoji not found'];
+        }
+
+        //Write message to response interface
+        $response->getBody()->write(json_encode($result));
+
+        //Return response
+        return $response;
+    }
+
+    /**
+     * Index route for getting all Emojis.
+     *
+     * @param ServerRequestInterface ServerRequestInterface $request  PSR-7 standard for receiving client request
+     * @param ResponseInterface      ResponseInterface      $response PSR-& standard for sending server response
+     *
+     * @return ResponseInterface HTTP response of client request
+     */
+    public function create(ServerRequestInterface $request, ResponseInterface $response)
+    {
+        //Get Emoji data from request
+        $data = $request->getParsedBody();
+
+        //Extract keywords from data
+        $keywords = $data['keywords'];
+
+        //Decode token to get object of data
+        $decodedToken = Auth::decodeToken($request->getHeader('Authorization')[0], $request);
+
+        //Check if user exist with the user id
+        $user = User::find($decodedToken->data->uid);
+
+        //Generate emoji data to return
+        $emoji = [
+            'name'     => $data['name'],
+            'char'     => $data['char'],
+            'category' => $data['category'],
+        ];
+
+        DB::transaction(function () use ($user, $emoji, $keywords) {
```

Comment:
**Code Quality - Good**
Good design choice -- instead of using a loop for multiple inserts.